src/nonunity/actions/NEML2Action.C

```diff
@@ -0,0 +1,142 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifdef NEML2_ENABLED
+
+#include "neml2/misc/utils.h"
+#include "neml2/misc/parser_utils.h"
+#include "neml2/base/HITParser.h"
+#include "NEML2Action.h"
+#include "NEML2Utils.h"
+#include "FEProblem.h"
+#include "Factory.h"
+
+registerMooseAction("BlackBearApp", NEML2Action, "parse_neml2");
+registerMooseAction("BlackBearApp", NEML2Action, "add_material");
+registerMooseAction("BlackBearApp", NEML2Action, "add_user_object");
+
+InputParameters
+NEML2Action::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Action to parse and setup NEML2 objects.");
+  params.addRequiredParam<FileName>("input",
+                                    "Path to the NEML2 input file containing the NEML2 model(s)");
+  params.addRequiredParam<std::string>(
+      "model",
+      "Name of the NEML2 model, i.e., the string inside the brackets [] in the NEML2 input file "
+      "that corresponds to the model you want to use.");
+  params.addParam<bool>("verbose",
+                        true,
+                        "Whether to print additional information about the NEML2 model at the "
+                        "beginning of the simulation");
+  params.addParam<std::vector<VariableName>>(
+      "temperature", std::vector<VariableName>{"0"}, "Coupled temperature");
+
+  MooseEnum mode("ELEMENT ALL PARSE_ONLY", "ELEMENT");
+  params.addParam<MooseEnum>(
+      "mode",
+      mode,
+      "Mode of operation. ELEMENT: Perform constitutive update element-by-element. ALL: Perform "
```

Comment:
Isn't there some cool new feature that lets us describe the enum entries and format them nicely in the documentation? I wasn't able to immediately find an example, but it would be nice to use that.

Reply:
Uhh I'm also having trouble finding an example for that.

Reply:
Thanks. I just added them.